internal/controller/nonadminbackup_controller.go

```diff
@@ -279,32 +270,22 @@ func (r *NonAdminBackupReconciler) UpdateSpecStatus(ctx context.Context, logrLog
 		veleroBackupLogger.Info("VeleroBackup successfully created")
 	}
 
-	updated := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseCreated)
-	if updated {
-		if err := r.Status().Update(ctx, nab); err != nil {
-			logger.Error(err, phaseUpdateError)
-			return true, false, err
-		}
-
-		logger.V(1).Info(phaseUpdateRequeue)
-		return false, true, nil
-	}
-
-	updated = meta.SetStatusCondition(&nab.Status.Conditions,
+	updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseCreated)
+	updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
 		metav1.Condition{
 			Type:    string(nacv1alpha1.NonAdminConditionQueued),
 			Status:  metav1.ConditionTrue,
 			Reason:  "BackupScheduled",
 			Message: "Created Velero Backup object",
 		},
 	)
-	if updated {
+	if updatedPhase || updatedCondition {
 		if err := r.Status().Update(ctx, nab); err != nil {
 			logger.Error(err, conditionUpdateError)
 			return true, false, err
 		}
-
-		logger.V(1).Info(conditionUpdateRequeue)
+		// TODO message
+		logger.V(1).Info(phaseUpdateRequeue)
 		return false, true, nil
 	}
```

Comment:
here, I could already update veleroBackup name and namespace in NAB status and exit, instead of Requeue

what you prefer?

Reply:
In images

It is currently like this
```mermaid
%%{init: {'theme':'dark'}}%%
graph
event1{{predicate: Create event NAB}} ==>

reconcileStart1[/Reconcile start\] ==>

phaseNew["`status.phase to **New**`"] -. Requeue: false, err .-> reconcileStart1
phaseNew ==>

reconcileExit1[\Requeue: true, nil/]
reconcileExit1 ==> valid
reconcileExit1 ==> invalid

subgraph valid ["`valid spec`"]
  direction TB
  reconcileStart2[/Reconcile start\] ==>

  conditionsAcceptedTrue["`status.conditions[Accepted] to **True**`"] -. Requeue: false, err .- reconcileStart2
  conditionsAcceptedTrue ==>

  reconcileExit2[\Requeue: true, nil/] ==>

  reconcileStart3[/Reconcile start\] ==>

  createVeleroBackup([Create Velero Backup]) -. Requeue: false, err .- reconcileStart3
  createVeleroBackup ==>

  phaseCreated["`status.phase: **Created**
  status.conditions[Queued] to **True**`"] -. Requeue: false, err .- reconcileStart3
  phaseCreated ==>

  reconcileExit3[\Requeue: true, nil/] ==>

  reconcileStart4[/Reconcile start\] ==>

  reconcileStart5[/Reconcile start\] ==>

  conditionsVeleroBackupName["`status.conditions.veleroBackup.name and status.conditions.veleroBackup.namespace`"] -. Requeue: false, err .- reconcileStart5
  conditionsVeleroBackupName ==>

  reconcileExit5[\Requeue: false, nil/] ~~~

  event2{{predicate: Update event VeleroBackup}} ==>

  reconcileStart6[/Reconcile start\] ==> conditionsVeleroBackupStatus

  conditionsVeleroBackupStatus["`status.conditions.veleroBackup.status`"] -. Requeue: false, err .- reconcileStart6
  conditionsVeleroBackupStatus ==>

  reconcileExit6[\Requeue: false, nil/]
end

subgraph invalid ["`invalid spec`"]
  direction TB
  reconcileStartInvalid1[/Reconcile start\] ==>

  phaseBackingOff["`status.phase to **BackingOff**
  status.conditions[Accepted] to **False**`"] -. Requeue: false, err .-> reconcileStartInvalid1
  phaseBackingOff ==>

  reconcileExitInvalid1[\Requeue: false, TerminalError/]
end
```

I want it to be like this
```mermaid
%%{init: {'theme':'dark'}}%%
graph
event1{{predicate: Create event NAB}} ==>

reconcileStart1[/Reconcile start\] ==>

phaseNew["`status.phase to **New**`"] -. Requeue: false, err .-> reconcileStart1
phaseNew ==>

reconcileExit1[\Requeue: true, nil/]
reconcileExit1 ==> valid
reconcileExit1 ==> invalid

subgraph valid ["`valid spec`"]
  direction TB
  reconcileStart2[/Reconcile start\] ==>

  conditionsAcceptedTrue["`status.conditions[Accepted] to **True**`"] -. Requeue: false, err .- reconcileStart2
  conditionsAcceptedTrue ==>

  reconcileExit2[\Requeue: true, nil/] ==>

  reconcileStart3[/Reconcile start\] ==>

  createVeleroBackup([Create Velero Backup]) -. Requeue: false, err .- reconcileStart3
  createVeleroBackup ==>

  phaseCreated["`status.phase: **Created**
  status.conditions[Queued] to **True**
  status.conditions.veleroBackup.name
  status.conditions.veleroBackup.namespace`"] -. Requeue: false, err .- reconcileStart3
  phaseCreated ==>

  reconcileExit5[\Requeue: false, nil/] ~~~

  event2{{predicate: Update event VeleroBackup}} ==>

  reconcileStart6[/Reconcile start\] ==> conditionsVeleroBackupStatus

  conditionsVeleroBackupStatus["`status.conditions.veleroBackup.status`"] -. Requeue: false, err .- reconcileStart6
  conditionsVeleroBackupStatus ==>

  reconcileExit6[\Requeue: false, nil/]
end

subgraph invalid ["`invalid spec`"]
  direction TB
  reconcileStartInvalid1[/Reconcile start\] ==>

  phaseBackingOff["`status.phase to **BackingOff**
  status.conditions[Accepted] to **False**`"] -. Requeue: false, err .-> reconcileStartInvalid1
  phaseBackingOff ==>

  reconcileExitInvalid1[\Requeue: false, TerminalError/]
end
```

Reply:
yeah makes sense to do that as we are adding `Created` phase to NAB and Create Velero call also get executed.

Reply:
done